exercises/concept/authentication-system/AuthenticationSystemTests.cs

```diff
@@ -24,4 +35,22 @@ public void GetDevelopers()
         bool?[] expected = { true, true, true };
         Assert.Equal(expected, actual);
     }
+
+    [Fact]
+    [Task(5)]
+    public void CheckDevelopersCannotBeTampered()
+    {
+        var authenticator = new Authenticator(new Identity { EyeColor = "green", Email = "admin@ex.ism" });
+        IDictionary<string, Identity> devs = authenticator.GetDevelopers();
+
+        Identity tamperedDev = new Identity { EyeColor = "grey", Email = "anders@hack.ed" };
+        try
+        {
+            devs["Anders"] = tamperedDev;
+            Assert.True(false, "Unexpected change to developers list.");
+        } catch (NotSupportedException)
+        {
+            Assert.True(true);            
+        }
```

Comment:
This should also work I think:

```suggestion
        Assert.Throws<NotSupportedException>(() => devs["Anders"] = tamperedDev);
```

Reply:
Changes done.